CRM/Contribute/Form/ContributionView.php

```diff
@@ -44,6 +44,16 @@ public function preProcess() {
     $values = (array) $contribution;
     $contributionStatus = CRM_Core_PseudoConstant::getName('CRM_Contribute_BAO_Contribution', 'contribution_status_id', $values['contribution_status_id']);
 
+    if (!isset($this->get_template_vars()['hookDiscount'])) {
+      $this->assign('hookDiscount', ['message' => '']);
```

Comment:
urgh but ok

Reply:
Yeah. Looking at it now maybe another way would be to just add hookDiscount in the expected smarty vars below, and then update the `if` in the template so it's `{if $hookDiscount && $hookDiscount.message}` instead of `{if $hookDiscount.message}`. I think I still have one more round before the other PR will pass so I will probably do that.